BeeSwiftTests/GoalViewModelTests.swift

```diff
@@ -0,0 +1,86 @@
+//
+//  GoalViewModelTests.swift
+//  BeeSwiftTests
+//
+//  Created by krugerk on 2024-11-25.
+//
+
+import Testing
+
+@testable import BeeSwift
+@testable import BeeKit
+
+struct GoalViewModelTests {
+    
+    @Test func initialStepperIsMinusOneWhenSubmissionDateIsAfterMidnightAndBeforeDeadline() async throws {
+        let goal = Self.makeGoalWithDeadline(3600 * 3)
+        let viewModel = GoalViewModel(goal: goal)
+        let submissionDate = Calendar.current.date(bySettingHour: 1, minute: 30, second: 0, of: Date())!
+        let actual = viewModel.initialDateStepperValue(date: submissionDate)
+        #expect(actual == -1)
+    }
+    
+    @Test func initialStepperIsZeroWhenSubmissionDateIsBeforeMidnightAndBeforeDeadline() async throws {
+        let goal = Self.makeGoalWithDeadline(0)
+        let viewModel = GoalViewModel(goal: goal)
+        let submissionDate = Calendar.current.date(bySettingHour: 20, minute: 30, second: 0, of: Date())!
+        let actual = viewModel.initialDateStepperValue(date: submissionDate)
+        #expect(actual == 0)
+    }
+
+    @Test func initialStepperIsPlusOneWhenSubmissionDateIsAfterDeadline() async throws {
```

Comment:
Can we give the reader some context on _why_ PlusOne is the correct value in this context? Either via the name, or comments.